ahnlich/ai/src/engine/ai/models.rs

```diff
@@ -120,19 +120,18 @@ impl Model {
     #[tracing::instrument(skip(self))]
     pub fn model_ndarray(
         &self,
-        storeinput: &ModelInput,
+        storeinput: &Vec<ModelInput>,
         action_type: &InputAction,
-    ) -> Result<StoreKey, AIProxyError> {
-        match &self.provider {
-            ModelProviders::FastEmbed(provider) => {
-                let embedding = provider.run_inference(storeinput, action_type)?;
-                Ok(StoreKey(<Array1<f32>>::from(embedding)))
-            }
-            ModelProviders::ORT(provider) => {
-                let embedding = provider.run_inference(storeinput, action_type)?;
-                Ok(StoreKey(<Array1<f32>>::from(embedding)))
-            }
-        }
+    ) -> Result<Vec<StoreKey>, AIProxyError> {
+        let embeddings = match &self.provider {
+            ModelProviders::FastEmbed(provider) => provider.run_inference(storeinput, action_type)?,
+            ModelProviders::ORT(provider) => provider.run_inference(storeinput, action_type)?
+        };
+        let store_keys = embeddings
```

Comment:
If we do [this](https://github.com/deven96/ahnlich/pull/134/files#r1815264646) we can get rid of this